apps/pkeyutl.c

```diff
@@ -547,13 +567,19 @@ static EVP_PKEY_CTX *init_ctx(const char *kdfalg, int *pkeysize,
                 goto end;
             }
         }
-        ctx = EVP_PKEY_CTX_new_id(kdfnid, impl);
+        if (libctx != NULL)
+            ctx = EVP_PKEY_CTX_new_from_name(libctx, kdfalg, propq);
+        else
+            ctx = EVP_PKEY_CTX_new_id(kdfnid, impl);
```

Comment:
It is fine for the libctx to be NULL..
That then just uses the default context..
The same is true for the engine pointer also.

It could at best do
```
if (impl == NULL)
   new_from_pkey()
else
  new( , impl)
```